### PR TITLE
SNOW-540086: cover critical areas test code coverage for logger, dsn and connection

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-lb**
### Description
Increase log.go coverage to 82.7% and cover missing critical areas in dsn.go and connection.go.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

File: connection_test.go [hunk 1]
```diff
@@ -657,3 +657,42 @@ func TestQueryContextError(t *testing.T) {
...
+		t.Error(err)
+	}
+	sc, err := buildSnowflakeConn(ctx, *config)
+	if err != nil {
+		t.Error(err)
+	}
+	if err = authenticateWithConfig(sc); err != nil {
+		t.Error(err)
+	}
+	tx, _ := sc.Begin()
```
Comment (sfc-gh-pfus) [inline connection_test.go:694]: Shouldn't we close a transaction here?

File: connection_test.go [hunk 2]
```diff
@@ -657,3 +657,42 @@ func TestQueryContextError(t *testing.T) {
...
+	}
+}
+
+func TestBeginCreatesTransaction(t *testing.T) {
+	ctx := context.Background()
+	config, err := ParseDSN(dsn)
+	if err != nil {
+		t.Error(err)
+	}
+	sc, err := buildSnowflakeConn(ctx, *config)
```
Comment (sfc-gh-pfus) [inline connection_test.go:687]: Shouldn't we call `Close` at the end?

File: connection_test.go [hunk 3]
```diff
@@ -657,3 +657,42 @@ func TestQueryContextError(t *testing.T) {
...
 	}
 }
+
+func TestPrepareQuery(t *testing.T) {
+	ctx := context.Background()
+	config, err := ParseDSN(dsn)
+	if err != nil {
+		t.Error(err)
+	}
+	sc, err := buildSnowflakeConn(ctx, *config)
```
Comment (sfc-gh-pfus) [inline connection_test.go:667]: Shouldn't we call Close at the end?

**Pull request merged**